telemetry/data_window.go

```diff
@@ -80,6 +110,123 @@ var (
 	subWindowsLock    = sync.RWMutex{}
 )
 
+func getSQLSum(sqlTypeDeta *sqlType) int64 {
+	result := sqlTypeDeta.Use
+	result += sqlTypeDeta.Show
+	result += sqlTypeDeta.Begin
+	result += sqlTypeDeta.Commit
+	result += sqlTypeDeta.Rollback
+	result += sqlTypeDeta.Insert
+	result += sqlTypeDeta.Replace
+	result += sqlTypeDeta.Delete
+	result += sqlTypeDeta.Update
+	result += sqlTypeDeta.Select
+	result += sqlTypeDeta.Other
+	return result
+}
+
+func readSQLMetric(timepoint time.Time, SQLResult *sqlUsageData) error {
+	ctx := context.TODO()
+	promQL := "sum(tidb_executor_statement_total{}) by (instance,type)"
+	result, err := querySQLMetric(ctx, timepoint, promQL)
+	if err != nil {
+		if err1, ok := err.(*promv1.Error); ok {
+			return errors.Errorf("query metric error, msg: %v, detail: %v", err1.Msg, err1.Detail)
+		}
+		return errors.Errorf("query metric error: %v", err.Error())
+	}
+
+	anylisSQLUsage(result, SQLResult)
+	return nil
+}
+
+func querySQLMetric(ctx context.Context, queryTime time.Time, promQL string) (result pmodel.Value, err error) {
+	// Add retry to avoid network error.
+	var prometheusAddr string
+	for i := 0; i < 5; i++ {
+		//TODO: the prometheus will be Integrated into the PD, then we need to query the prometheus in PD directly, which need change the quire API
+		prometheusAddr, err = infosync.GetPrometheusAddr()
+		if err == nil || err == infosync.ErrPrometheusAddrIsNotSet {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	if err != nil {
+		return nil, err
+	}
+	promClient, err := api.NewClient(api.Config{
+		Address: prometheusAddr,
+	})
+	if err != nil {
+		return nil, err
+	}
+	promQLAPI := promv1.NewAPI(promClient)
+	ctx, cancel := context.WithTimeout(ctx, promReadTimeout)
+	defer cancel()
+	// Add retry to avoid network error.
+	for i := 0; i < 5; i++ {
+		result, _, err = promQLAPI.Query(ctx, promQL, queryTime)
+		if err == nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return result, err
+}
+
+func anylisSQLUsage(promResult pmodel.Value, SQLResult *sqlUsageData) {
+	switch promResult.Type() {
+	case pmodel.ValVector:
+		matrix := promResult.(pmodel.Vector)
+		for _, m := range matrix {
+			v := m.Value
+			fillSQLResult(m.Metric, v, SQLResult)
+		}
+	}
+}
+
+func fillSQLResult(metric pmodel.Metric, pair pmodel.SampleValue, SQLResult *sqlUsageData) {
+	v := ""
+	if metric != nil {
+		v = string(metric[pmodel.LabelName("type")])
+	}
+	if len(v) == 0 {
+		v = "total"
+	}
+	var record int64
+	if math.IsNaN(float64(pair)) {
+		record = 0
+	} else {
+		record = int64(float64(pair))
+	}
+	switch v {
+	case "total":
```

Comment:
These labels are likely to be changed. Either write a logic that resist such change, or write unit tests to ensure that they are unchanged.

Reply:
We'd better add logic tho resit such change.

By the way, when will the change happen?

Can we do it next sprint?